### PR TITLE
[apple-authentication][docs] fix error code, other small tweaks

### LINKED ISSUE #20917: [docs] [expo-apple-authentication] Incorrect error code for cancelled requests
### Summary

I've been using the main branch of `expo-apple-authentication` due to #20600, as my project requires compatibility with Fabric. The current docs ([apple-authentication.mdx, line 92](https://github.com/expo/expo/blob/main/docs/pages/versions/unversioned/sdk/apple-authentication.mdx)) show the wrong error code for when a user cancels their request: in reality, `ERR_REQUEST_CANCELED` is returned instead of `ERR_CANCELED`.

I'm not entirely sure if `err.code` behavior is consistent across Fabric and non-Fabric builds, but I figured it'd be worthwhile to mention diverging error behaviors in the docs should that not be the case.

### Link to the related docs page

https://github.com/expo/expo/blob/main/docs/pages/versions/unversioned/sdk/apple-authentication.mdx

### Anything else?

I tested this with an iPhone 11 Pro (MWC32BZ/A, iOS 16.0.2), React 18.2, React Native 0.71.1, and Expo 47.0.13. I'm running `expo-apple-authentication`@#20675, the latest relevant commit on the `main` branch as of today.

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes #20917
Refs #20600

# How

Correct the error name for the latest package version (after rewrite to Expo Modules).

Update JSDoc comments, update unversioned doc page, regenerate API data.

Besides those changes I have also tweaked the package description and keywords.

# Test Plan

Docs changes has been verified by running the website app locally.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]

File: docs/pages/versions/unversioned/sdk/apple-authentication.mdx [hunk 1]
```diff
@@ -143,10 +143,15 @@ import * as AppleAuthentication from 'expo-apple-authentication';
...
-| ERR_APPLE_AUTHENTICATION_UNAVAILABLE    | Apple authentication is unavailable on the device.                                                     |
-| ERR_CANCELED                            | The user canceled the sign-in request from the system modal.                                           |
+Most of the error codes matches the official [Apple Authorization errors](https://developer.apple.com/documentation/authenticationservices/asauthorizationerror/code).
+
+| Code                        | Description                                                                                          |
+| --------------------------- | ---------------------------------------------------------------------------------------------------- |
+| ERR_INVALID_OPERATION       | An invalid authorization operation has been performed.                                               |
+| ERR_INVALID_RESPONSE        | The authorization request received an invalid response.                                              |
+| ERR_INVALID_SCOPE           | An invalid [`AppleAuthenticationScope`](#appleauthenticationappleauthenticationscope) was passed in. |
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/apple-authentication.mdx:152]:
```suggestion
| ERR_INVALID_SCOPE           | An invalid [`AppleAuthenticationScope`](#appleauthenticationscope) was passed in. |
```

**Review by amandeepmittal — APPROVED** [review]
Thanks for updating this! Lgtm 👍 
One small suggestion.

File: docs/pages/versions/unversioned/sdk/apple-authentication.mdx (hunk 1 above)
Comment (Simek) [reply docs/pages/versions/unversioned/sdk/apple-authentication.mdx:152]: Nice catch, thank you! 🙏

File: docs/pages/versions/unversioned/sdk/apple-authentication.mdx [hunk 2]
```diff
@@ -143,10 +143,15 @@ import * as AppleAuthentication from 'expo-apple-authentication';
...
+| ERR_REQUEST_CANCELED        | User canceled the authorization attempt.                                                             |
+| ERR_REQUEST_FAILED          | The authorization request failed. See the error message for an additional information.               |
+| ERR_REQUEST_NOT_HANDLED     | The authorization request wasn’t correctly handled.                                                  |
+| ERR_REQUEST_NOT_INTERACTIVE | The authorization request isn’t interactive.                                                         |
+| ERR_REQUEST_UNKNOWN         | The authorization attempt failed for an unknown reason.                                              |
```
Comment (kbrandwijk) [inline docs/pages/versions/unversioned/sdk/apple-authentication.mdx:157]:
```suggestion
| ERR_REQUEST_UNKNOWN         | The authorization request failed for an unknown reason.                                              |
```
For consistency

File: docs/pages/versions/unversioned/sdk/apple-authentication.mdx [hunk 3]
```diff
@@ -143,10 +143,15 @@ import * as AppleAuthentication from 'expo-apple-authentication';
...
-| ERR_CANCELED                            | The user canceled the sign-in request from the system modal.                                           |
+Most of the error codes matches the official [Apple Authorization errors](https://developer.apple.com/documentation/authenticationservices/asauthorizationerror/code).
+
+| Code                        | Description                                                                                          |
+| --------------------------- | ---------------------------------------------------------------------------------------------------- |
+| ERR_INVALID_OPERATION       | An invalid authorization operation has been performed.                                               |
+| ERR_INVALID_RESPONSE        | The authorization request received an invalid response.                                              |
+| ERR_INVALID_SCOPE           | An invalid [`AppleAuthenticationScope`](#appleauthenticationappleauthenticationscope) was passed in. |
+| ERR_REQUEST_CANCELED        | User canceled the authorization attempt.                                                             |
```
Comment (kbrandwijk) [inline docs/pages/versions/unversioned/sdk/apple-authentication.mdx:153]:
```suggestion
| ERR_REQUEST_CANCELED        | User canceled the authorization request.                                                             |
```
For consistency.

**Pull request merged**